New check: Javadoc leading asterisk presence

I have read check documentation: https://checkstyle.sourceforge.io/checks/javadoc/javadocleadingasteriskalign.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

From: https://google.github.io/styleguide/javaguide.html#s7.1.1-javadoc-multi-line
> The basic formatting of Javadoc blocks is as seen in this example:
``` java
/**
 * Multiple lines of Javadoc text are written here,
 * wrapped normally...
 */
public int method(String p1) { ... }
```


Code: 
```java
class Test {
  /**
   Prepare the long long long long long long long long long long long long long long inputs.
   Prepare the inputs.
   Prepare the inputs.
   */
  void doSomething(int x) {
    int y = x + 2;
  }
}
```
Cli: 
```cmd
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml Test.java 
Starting audit...
Audit done.
```

Formatter:
```cmd
$ java -jar google-java-format-1.28.0-all-deps.jar Test.java > FormattedCode.java
$ diff -Naru Test.java FormattedCode.java
--- Test.java   2025-08-26 17:37:11.180988300 +0800
+++ FormattedCode.java  2025-08-26 17:37:19.101166500 +0800
@@ -1,10 +1,9 @@
 class Test {
   /**
-   Prepare the long long long long long long long long long long long long long long inputs.
-   Prepare the inputs.
-   Prepare the inputs.
+   * Prepare the long long long long long long long long long long long long long long inputs.
+   * Prepare the inputs. Prepare the inputs.
    */
   void doSomething(int x) {
     int y = x + 2;
   }
-}
\ No newline at end of file
+}
```

---

We need a new check to enforce that all lines in a Javadoc comment start with a leading *.
This should be similar in style to `JavadocLeadingAsteriskAlign`, but focused on presence rather than alignment.
The new check could be named `JavadocLeadingAsteriskPresence` (to stay parallel with `JavadocLeadingAsteriskAlign`).

---

@romani , We have this check with another name, `JavadocMissingLeadingAsteriskCheck`


```
D:\CS\test
cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocMissingLeadingAsteriskCheck"/>
    </module>
</module>
D:\CS\test
cat src/Test.java
class Test {
  /**
   Prepare the long long long long long long long long long long long long long long inputs.
   Prepare the inputs.
   Prepare the inputs.
   */
  void doSomething(int x) {
    int y = x + 2;
  }
}
D:\CS\test
java  -jar checkstyle-10.22.0-SNAPSHOT-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:3: Javadoc line should start with leading asterisk. [JavadocMissingLeadingAsterisk]
[ERROR] D:\CS\test\src\Test.java:4: Javadoc line should start with leading asterisk. [JavadocMissingLeadingAsterisk]
[ERROR] D:\CS\test\src\Test.java:5: Javadoc line should start with leading asterisk. [JavadocMissingLeadingAsterisk]
Audit done.
Checkstyle ends with 3 errors.
```

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputCorrectJavadocLeadingAsteriskAlignment.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputCorrectJavadocLeadingAsteriskAlignment.java
@@ -20,13 +20,15 @@ public class InputCorrectJavadocLeadingAsteriskAlignment {
   /** */ // violation "Summary javadoc is missing."
   private String str1;
 
+  // violation 2 lines below "Javadoc line should start with leading asterisk."
   /**
-    No leading asterisk present. False negative until #17778, Subproblem 2.
+    No leading asterisk present.
    */
   public InputCorrectJavadocLeadingAsteriskAlignment() {}
 
-  // violation 2 lines below "Summary javadoc is missing."
-  // violation 2 lines below "Javadoc tag '@param' should be preceded with an empty line."
+  // violation 3 lines below "Summary javadoc is missing."
+  // violation 3 lines below "Javadoc tag '@param' should be preceded with an empty line."
+  // False negative for below javadoc, until #18271
   /***
    * @param a testing....
    */
@@ -35,7 +37,7 @@ public class InputCorrectJavadocLeadingAsteriskAlignment {
   /*************************************************
    *** @param str testing.....
    **********************************/
-  // False negative for above javadoc, until #17778, Subproblem 1 & 3.
+  // False negative for above javadoc, until #18271, #18273.
   public InputCorrectJavadocLeadingAsteriskAlignment(String str) {}
 
   /** * */ // violation "Summary javadoc is missing."
@@ -49,47 +51,51 @@ public class InputCorrectJavadocLeadingAsteriskAlignment {
    */
   private void foo() {}
 
-  /** Opening tag should be alone on line. // False negative until #17778, Subproblem 1.
+  // violation below "Javadoc content should start from the next line."
+  /** Opening tag should be alone on line.
    * This method does nothing.
    */
   private void foo2() {}
 
   /**
    * Javadoc for foo3.
-   Closing tag should be alone on line. */ // False negative until #17778, Subproblem 3.
+   Closing tag should be alone on line. */ // False negative until #18273.
+  // violation above "Javadoc line should start with leading asterisk."
   private void foo3() {
     // foo2 code goes here
   }
 
   /**
    * Javadoc for enum.
-   * */ // False negative until #17778, Subproblem 3.
+   * */ // False negative until #18273.
   private enum CorrectJavadocEnum {
     // violation 2 lines below "Summary javadoc is missing."
-    // False negative until #17778, Subproblem 2.
+    // violation 2 lines below "Javadoc line should start with leading asterisk."
     /**
 
      */
     ONE,
 
-    // False negative until #17778, Subproblem 2.
+    // violation 2 lines below "Javadoc line should start with leading asterisk."
     /**
      Not allowed. No leading asterisk present.
      */
     TWO,
 
-    // False negative until #17778, Subproblem 1 & 2.
+    // violation 2 lines below "Javadoc content should start from the next line."
+    // violation 2 lines below "Javadoc line should start with leading asterisk."
     /** Not allowed. Opening javadoc tag should be alone on line.
 
      */
     THREE,
 
-    // False negative until #17778, Subproblem 3.
+    // False negative until #18273.
     /**
      * Not allowed. Closing javadoc tag should be alone on line. */
     FOUR,
 
-    // violation below "Summary javadoc is missing."
+    // violation 2 lines below "Summary javadoc is missing."
+    // violation 2 lines below "Javadoc line should start with leading asterisk."
     /**
 
      */

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedCorrectJavadocLeadingAsteriskAlignment.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedCorrectJavadocLeadingAsteriskAlignment.java
@@ -22,11 +22,12 @@ public class InputFormattedCorrectJavadocLeadingAsteriskAlignment {
   // violation above "Summary javadoc is missing."
   private String str1;
 
-  /** No leading asterisk present. False negative until #17778, Subproblem 2. */
+  /** No leading asterisk present. */
   public InputFormattedCorrectJavadocLeadingAsteriskAlignment() {}
 
-  // violation 2 lines below "Summary javadoc is missing."
-  // violation 2 lines below "Javadoc tag '@param' should be preceded with an empty line."
+  // violation 3 lines below "Summary javadoc is missing."
+  // violation 3 lines below "Javadoc tag '@param' should be preceded with an empty line."
+  // False negative for below javadoc, until #18271
   /***
    * @param a testing....
    */
@@ -35,7 +36,7 @@ public class InputFormattedCorrectJavadocLeadingAsteriskAlignment {
   /*************************************************
    *** @param str testing.....
    **********************************/
-  // False negative for above javadoc, until #17778, Subproblem 1 & 3.
+  // False negative for above javadoc, until #18271, #18273
   public InputFormattedCorrectJavadocLeadingAsteriskAlignment(String str) {}
 
   /** * */
@@ -49,35 +50,26 @@ public class InputFormattedCorrectJavadocLeadingAsteriskAlignment {
   /** This method does nothing. */
   private void foo() {}
 
-  /**
-   * Opening tag should be alone on line. // False negative until #17778, Subproblem 1. This method
-   * Does nothing.
-   */
+  /** Opening tag should be alone on line. This method does nothing. */
   private void foo2() {}
 
   /** Javadoc for foo3. Closing tag should be alone on line. */
-  // False negative until #17778, Subproblem 3.
   private void foo3() {
     // foo2 code goes here
   }
 
   /** Javadoc for enum. */
-  // False negative until #17778, Subproblem 3.
   private enum CorrectJavadocEnum {
-    // violation 2 lines below "Summary javadoc is missing."
-    // False negative until #17778, Subproblem 2.
+    // violation below "Summary javadoc is missing."
     /** */
     ONE,
 
-    // False negative until #17778, Subproblem 2.
     /** Not allowed. No leading asterisk present. */
     TWO,
 
-    // False negative until #17778, Subproblem 1 & 2.
     /** Not allowed. Opening javadoc tag should be alone on line. */
     THREE,
 
-    // False negative until #17778, Subproblem 3.
     /** Not allowed. Closing javadoc tag should be alone on line. */
     FOUR,
 

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedIncorrectJavadocLeadingAsteriskAlignment.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedIncorrectJavadocLeadingAsteriskAlignment.java
@@ -29,7 +29,7 @@ public class InputFormattedIncorrectJavadocLeadingAsteriskAlignment {
 
   /** Misaligned leading asterisk. Inner Class. */
   private static class Inner {
-    // No leading asterisk present. False-negative until #17778, Subproblem 2
+
     // violation below "Summary javadoc is missing."
     /** */
     private Object obj;
@@ -47,7 +47,6 @@ public class InputFormattedIncorrectJavadocLeadingAsteriskAlignment {
     /** */
     ONE,
 
-    // Closing tag should be alone on line. False negative until #17778, subproblem 3
     /** Wrong Alignment. */
     TWO
   }

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputIncorrectJavadocLeadingAsteriskAlignment.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputIncorrectJavadocLeadingAsteriskAlignment.java
@@ -59,8 +59,8 @@ public class InputIncorrectJavadocLeadingAsteriskAlignment {
     * Inner Class. */
   // violation above 'Leading asterisk has .* indentation .* 5, expected is 4.'
   private static class Inner {
-    // No leading asterisk present. False-negative until #17778, Subproblem 2
-    // violation below "Summary javadoc is missing."
+    // violation 2 lines below "Summary javadoc is missing."
+    // violation 2 lines below "Javadoc line should start with leading asterisk."
     /**
 
         */
@@ -86,7 +86,7 @@ public class InputIncorrectJavadocLeadingAsteriskAlignment {
     ONE,
 
 
-    // Closing tag should be alone on line. False negative until #17778, subproblem 3
+    // Closing tag should be alone on line. False negative until #18273
     /**
       * Wrong Alignment. */
     // violation above 'Leading asterisk has .* indentation .* 7, expected is 6'

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectJavadocParagraph.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectJavadocParagraph.java
@@ -40,6 +40,7 @@ class InputIncorrectJavadocParagraph {
     return false;
   }
 
+  // violation 5 lines below "Javadoc content should start from the next line."
   // violation 4 lines below 'Redundant <p> tag.'
   // 2 violations 4 lines below:
   //  '<p> tag should be placed immediately before the first word'
@@ -70,7 +71,8 @@ class InputIncorrectJavadocParagraph {
      */
     public static final byte NUL = 0;
 
-    // 2 violations 5 lines below:
+    // 3 violations 6 lines below:
+    //  'Javadoc content should start from the next line.'
     //  '<p> tag should be placed immediately before the first word'
     //  'Redundant <p> tag.'
     // violation 5 lines below '<p> tag should be placed immediately before the first word'

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule734nonrequiredjavadoc/InputInvalidJavadocPositionRecord.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule734nonrequiredjavadoc/InputInvalidJavadocPositionRecord.java
@@ -1,6 +1,7 @@
 package com.google.checkstyle.test.chapter7javadoc.rule734nonrequiredjavadoc;
 
-// violation below 'Javadoc comment is placed in the wrong location.'
+// violation 2 lines below 'Javadoc comment is placed in the wrong location.'
+// violation 2 lines below 'Javadoc content should start from the next line.'
 /** Odd javadoc */
 /** Valid javadoc.
  *

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -196,6 +196,8 @@
     <module name="MultipleVariableDeclarations"/>
     <module name="ArrayTypeStyle"/>
     <module name="JavadocLeadingAsteriskAlign"/>
+    <module name="JavadocMissingLeadingAsterisk"/>
+    <module name="JavadocContentLocation"/>
     <module name="MissingSwitchDefault"/>
     <module name="FallThrough"/>
     <module name="UpperEll"/>

--- a/src/site/xdoc/checks/javadoc/javadoccontentlocation.xml
+++ b/src/site/xdoc/checks/javadoc/javadoccontentlocation.xml
@@ -173,6 +173,10 @@ class Example2 {
             <a href="https://github.com/search?q=path%3Aconfig%20path%3A**%2Fcheckstyle-checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocContentLocation">
               Checkstyle Style</a>
           </li>
+          <li>
+            <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocContentLocation">
+              Google Style</a>
+          </li>
         </ul>
       </subsection>
 

--- a/src/site/xdoc/checks/javadoc/javadoccontentlocation.xml.template
+++ b/src/site/xdoc/checks/javadoc/javadoccontentlocation.xml.template
@@ -77,6 +77,10 @@
             <a href="https://github.com/search?q=path%3Aconfig%20path%3A**%2Fcheckstyle-checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocContentLocation">
               Checkstyle Style</a>
           </li>
+          <li>
+            <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocContentLocation">
+              Google Style</a>
+          </li>
         </ul>
       </subsection>
 

--- a/src/site/xdoc/checks/javadoc/javadocmissingleadingasterisk.xml
+++ b/src/site/xdoc/checks/javadoc/javadocmissingleadingasterisk.xml
@@ -103,6 +103,10 @@ class Example1 {}
             <a href="https://github.com/search?q=path%3Aconfig%20path%3A**%2Fcheckstyle-checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocMissingLeadingAsterisk">
               Checkstyle Style</a>
           </li>
+          <li>
+            <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocMissingLeadingAsterisk">
+              Google Style</a>
+          </li>
         </ul>
       </subsection>
 

--- a/src/site/xdoc/checks/javadoc/javadocmissingleadingasterisk.xml.template
+++ b/src/site/xdoc/checks/javadoc/javadocmissingleadingasterisk.xml.template
@@ -50,6 +50,10 @@
             <a href="https://github.com/search?q=path%3Aconfig%20path%3A**%2Fcheckstyle-checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocMissingLeadingAsterisk">
               Checkstyle Style</a>
           </li>
+          <li>
+            <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocMissingLeadingAsterisk">
+              Google Style</a>
+          </li>
         </ul>
       </subsection>
 

--- a/src/site/xdoc/google_style.xml
+++ b/src/site/xdoc/google_style.xml
@@ -2374,6 +2374,28 @@
                   <br />
                   <span class="wrapper inline">
                     <img
+                            src="images/ok_green.png"
+                            alt="" />
+                  </span>
+                  <a href="checks/javadoc/javadocmissingleadingasterisk.html#JavadocMissingLeadingAsterisk">
+                    JavadocMissingLeadingAsterisk</a>
+                  (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocMissingLeadingAsterisk">
+                  config</a>)
+                  <br />
+                  <br />
+                  <span class="wrapper inline">
+                    <img
+                            src="images/ok_blue.png"
+                            alt="" />
+                  </span>
+                  <a href="checks/javadoc/javadoccontentlocation.html#JavadocContentLocation">
+                    JavadocContentLocation</a>
+                  (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocContentLocation">
+                  config</a>)
+                  <br />
+                  <br />
+                  <span class="wrapper inline">
+                    <img
                             src="images/ok_blue.png"
                             alt="" />
                   </span>
@@ -2381,6 +2403,23 @@
                     InvalidJavadocPosition</a>
                   (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fgoogle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+InvalidJavadocPosition">
                   config</a>)
+                  <br />
+                  <br />
+                  Non covered:
+                  <br />
+                  1. There is a false negative related to Javadoc content that begins
+                  with a leading asterisk. It is addressed at:
+                  <a href="https://github.com/checkstyle/checkstyle/issues/18271">
+                    #18271
+                  </a>
+                  <br />
+                  <br />
+                  2. There is a false negative concerning Javadoc blocks where the
+                  closing <code>*/</code> does not appear on its own line.
+                  It is addressed at:
+                  <a href="https://github.com/checkstyle/checkstyle/issues/18273">
+                    #18273
+                  </a>
                 </td>
                 <td>
                   <a href="https://github.com/checkstyle/checkstyle/tree/master/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform">

